cardano-testnet/test/Spec/Cli/Alonzo/LeadershipSchedule.hs

```diff
@@ -0,0 +1,527 @@
+
+  void $ H.execCli' execConfig
+    [ "query", "tip"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--out-file", work </> "current-tip-2.json"
+    ]
+
+  tip2JSON <- H.leftFailM . H.readJsonFile $ work </> "current-tip-2.json"
+  tip2 <- H.noteShowM $ H.jsonErrorFail $ J.fromJSON @QueryTipLocalStateOutput tip2JSON
+
+  currEpoch2 <- case mEpoch tip2 of
+    Nothing -> H.failMessage callStack "cardano-cli query tip returned Nothing for EpochNo"
+    Just currEpoch2 -> return currEpoch2
+
+  H.note_ $ "Current Epoch: " <> show currEpoch2
+
+  now <- H.noteShowIO DTC.getCurrentTime
+  deadline <- H.noteShow $ DTC.addUTCTime 90 now
+
+  H.assertByDeadlineMCustom "stdout does not contain \"until genesis start time\"" deadline $ do
+    H.threadDelay 1000000
+    void $ H.execCli' execConfig
+      [ "query", "tip"
+      , "--testnet-magic", show @Int testnetMagic
+      , "--out-file", work </> "current-tip-3.json"
+      ]
+
+    tip3JSON <- H.leftFailM . H.readJsonFile $ work </> "current-tip-3.json"
+    tip3 <- H.noteShowM $ H.jsonErrorFail $ J.fromJSON @QueryTipLocalStateOutput tip3JSON
+
+    currEpoch3 <- case mEpoch tip3 of
+      Nothing -> H.failMessage callStack "cardano-cli query tip returned Nothing for EpochNo"
+      Just currEpoch3 -> return currEpoch3
+
+    H.note_ $ "Current Epoch: " <> show currEpoch3
+    return (currEpoch3 > currEpoch2 + 1)
+
+  ledgerStateJson <- H.execCli' execConfig
+    [ "query", "ledger-state"
+    , "--cardano-mode"
+    , "--testnet-magic", show @Int testnetMagic
+    ]
+
+  H.note_ ledgerStateJson
+
+  H.note_ "Done"
+
+  let poolVrfSkey = TR.poolNodeKeysVrfSkey $ TR.poolNodeKeys poolNode1
+  scheduleFile <- H.noteTempFile tempAbsPath "schedule.log"
+
+  void $ H.execCli' execConfig
+    [ "query", "leadership-schedule"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--genesis", TC.shelleyGenesisFile tr
+    , "--stake-pool-id", stakePoolId
+    , "--vrf-signing-key-file", poolVrfSkey
+    , "--out-file", scheduleFile
+    , "--current"
+    ]
+
+  scheduleJson <- H.leftFailM $ H.readJsonFile scheduleFile
+
+  expectedLeadershipSlotNumbers <- H.noteShowM $ fmap (fmap slotNumber) $ H.leftFail $ J.parseEither (J.parseJSON @[LeadershipSlot]) scheduleJson
+
+  H.assert $ not (L.null expectedLeadershipSlotNumbers)
+
+  leadershipDeadline <- H.noteShowM $ DTC.addUTCTime 90 <$> H.noteShowIO DTC.getCurrentTime
+
+  H.assertByDeadlineMCustom "Leader schedule is correct" leadershipDeadline $ do
+    leaderSlots <- H.getRelevantLeaderSlots (TR.poolNodeStdout poolNode1) (minimum expectedLeadershipSlotNumbers)
+    maxSlotExpected <- H.noteShow $ maximum expectedLeadershipSlotNumbers
+    maxActualSlot <- H.noteShow $ maximum leaderSlots
+    return $ maxActualSlot >= maxSlotExpected
+
+  leaderSlots <- H.getRelevantLeaderSlots (TR.poolNodeStdout poolNode1) (minimum expectedLeadershipSlotNumbers)
+
+  H.assert $ L.length (expectedLeadershipSlotNumbers \\ leaderSlots) <= 1
```

Comment:
Can you leave a comment as to why it's acceptable for the difference to be <= 1?

Reply:
Done!